### ISSUE TITLE
Use Jib Core in jib-gradle-plugin and jib-maven-plugin.

### ISSUE THREAD

**Issue by coollog** [GoogleContainerTools/jib#1118]
- [ ] Use Jib Core API in `jib-gradle-plugin`
- [ ] Use Jib Core API in `jib-maven-plugin`

I will work on the `jib-gradle-plugin` side and it'd be great if someone else takes the `jib-maven-plugin` side to get a feel of the new library.

**Comment by TadCordle** [comment]
I could take maven

**Comment by loosebazooka** [comment]
Assign yourself or edit the first comment to indicate ownership.

**Comment by coollog** [comment]
Actually, I was working on the Gradle side and there might be some overlap in changes that might affect Maven side (changes in `plugins-common`). It might be better to wait until I push the Gradle changes and work on top of those changes for the Maven side - unless you've gotten far in it already?

**Comment by TadCordle** [comment]
Nope, haven't started yet; still working on the cache stuff. I can wait until you've made progress.

**Comment by coollog** [comment]
@TadCordle I put up #1119, so working off of that should be good. The main thing to note is to switch maven to use `NBuildStepsRunner`, which will replace `BuildStepsRunner` once everything is changed over.
